app/code/Magento/Cms/Model/Wysiwyg/Images/PrepareImage.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Cms\Model\Wysiwyg\Images;
+
+use Magento\Catalog\Helper\Data;
+use Magento\Cms\Helper\Wysiwyg\Images as ImagesHelper;
+
+class PrepareImage
+{
+    /**
+     * @var ImagesHelper
+     */
+    private $imagesHelper;
+
+    /**
+     * @var Data
+     */
+    private $catalogHelper;
+
+    /**
+     * PrepareImage constructor.
+     * @param ImagesHelper $imagesHelper
+     * @param Data $catalogHelper
+     */
+    public function __construct(
+        ImagesHelper $imagesHelper,
+        Data $catalogHelper
+    ) {
+        $this->imagesHelper = $imagesHelper;
+        $this->catalogHelper = $catalogHelper;
+    }
+
+    /**
+     * @param array $data
+     * @return string
+     */
+    public function execute(array $data): string
```

Comment:
It's better to have the meaningful parameters:
```suggestion
    public function execute(string $endcodedFilename, int $storeId, bool $forceStaticPath, bool $renderAsTag): string
```